### ISSUE TITLE
plymouth not useable as a systempkg

### ISSUE THREAD

**Issue by konobi** [NixOS/nixpkgs#7400]
Currently installing plymouth will install libs and binaries, but the configuration and systemd files are not available in current-system.

**Comment by vcunat** [comment]
NixOS philosophy: installing stuff (e.g. putting into ``systemPackages``) doesn't by itself activate any services and such.

Adding it into ``systemd.packages`` should make systemd consider the units defined in the package, although I'm unsure that's enough in this particular case. Ideally, in future we might have ``services.plymouth.enable = true;``.

**Comment by konobi** [comment]
Ah, i had never seen systemd.packages before. but yeah, a service would be handy.

**Comment by konobi** [comment]
It appears that that wasn't sufficient =0(

**Comment by edolstra** [comment]
As far as I know, NixOS does not currently support Plymouth.

**Comment by cillianderoiste** [comment]
I never got plymouth to launch on boot, although I was able to launch it as a test from a running system and display the boot screens. I suspect it's possible to fiddle with the initrd to get it working, but it's beyond me (and sorry that I don't have a unified nick, I guess I should "rebrand" myself someday).

**Comment by konobi** [comment]
@edolstra what is involved with getting it supported? I'm okay to do a bunch of the work, I just need some pointers.

**Comment by cillianderoiste** [comment]
@konobi FWIW, I played around with boot.initrd.postDeviceCommands and postMountCommands. I think the current version of plymouth is out of date, so I would first try to update that and get the test examples working from a booted system. I have stumbled across code in nixpkgs which mentions plymouth (possibly disabling support for it, since we don't have it working yet), so that would also be worth a look.
